mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -842,22 +842,26 @@ def LLVM_GetActiveLaneMaskOp
 /// Create a call to Masked Load intrinsic.
 def LLVM_MaskedLoadOp : LLVM_OneResultIntrOp<"masked.load"> {
   let arguments = (ins LLVM_AnyPointer:$data, LLVM_VectorOf<I1>:$mask,
-                   Variadic<LLVM_AnyVector>:$pass_thru, I32Attr:$alignment);
+                   Variadic<LLVM_AnyVector>:$pass_thru, I32Attr:$alignment,
+                   DefaultValuedAttr<BoolAttr, "false">:$nontemporal);
```

Comment:
`UnitAttr` maybe? ... Update, no, this is fine

Reply:
... Also, if you're doing this, you might want to take a followup PR where you take `ScalarizeMaskedMemIntr` and have it copy over the `!nontemporal` (and other load-related metadata like alias analysis annotations) over to the loads

Reply:
I think you already have that code into `ScalarizeMaskedMemIntr` , because I see the flag propagated down to the loads.